feature/experimental/p4rt/ate_tests/base_p4rt/base_p4rt_test.go

```diff
@@ -280,19 +280,51 @@ func setupP4RTClient(ctx context.Context, args *testArgs) error {
 	return nil
 }
 
-// Function to compare and check if the expected table is present in RPC ReadResponse
-func verifyReadReceiveMatch(expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
+// Function to compare and check if the expected table is present in RPC ReadResponse.
+func verifyReadReceiveMatch(t *testing.T, expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
 
 	matches := 0
 	for _, table := range received_entry.Entities {
-		if diff := cmp.Diff(expected_update[0].Entity.Entity, table.Entity, protocmp.Transform()); diff != "" {
-			glog.Errorf("Table entry diff (-want +got): \n%s", diff)
-			continue
+		// Added a switch case for the captured table entry.
+		// Switch case to remove the counter_data and meter_config from table entry returned by p4rt server.
+		var tableMap map[string]any
+		switch entry := table.Entity.(type) {
+		case *p4_v1.Entity_TableEntry:
+			ent1, err := json.Marshal(entry)
+			if err != nil {
+				return fmt.Errorf("unable to convert table entry to json: %w", err)
+			}
+			var toMap map[string]any
+			err = json.Unmarshal([]byte(string(ent1)), &toMap)
+			if err != nil {
+				return fmt.Errorf("unable to unmarshal table entry to map: %w", err)
+			}
+			tableMap = toMap["TableEntry"].(map[string]any)
+			delete(tableMap, "meter_config")
+			delete(tableMap, "counter_data")
+		default:
+			t.Logf("Not a table entry: %v", entry)
+		}
+		ent2, err := json.Marshal(expected_update[0].Entity.Entity)
+		if err != nil {
+			return fmt.Errorf("unable to convert table entry to json: %w", err)
+		}
+		var toMap1 map[string]any
+
+		err = json.Unmarshal([]byte(string(ent2)), &toMap1)
+		if err != nil {
+			return fmt.Errorf("unable to convert table entry to json: %w", err)
 		}
-		matches++
+		tableMap1 := toMap1["TableEntry"].(map[string]any)
+
+		if cmp.Equal(tableMap, tableMap1) {
+			t.Logf("Table match succesful")
+			matches += 1
+		}
+
 	}
 	if matches == 0 {
-		return errors.New("match unsuccesful")
+		return fmt.Errorf("match unsuccesful since number of matches: %d", matches)
```

Comment:
No need to interpolate "matches" when you no it is zero. I'd probably just change it to "return errors.New("no matches found")

Reply:
Hi Greg,

This will be addressed in the next commit.